test/framework/resources/templates/reserve-capacity.yaml.tpl

```
@@ -21,6 +21,10 @@ spec:
       terminationGracePeriodSeconds: 5
       nodeSelector:
         worker.gardener.cloud/pool: {{ .WorkerPool }}
+      tolerations:
+      - key: {{ .TolerationKey }}
+        operator: "Exists"
+        effect: "NoSchedule"
```

This will add toleration for all keys to the deployment used in the other clusterautoscaler test (because the `.TolerationKey` is not set there).
Can you therefore wrap it in an `if` like this?
```suggestion
{{- if .TolerationKey }}
      tolerations:
      - key: {{ .TolerationKey }}
        operator: "Exists"
        effect: "NoSchedule"
{{- end }}
```

---

yes, sounds good.